public/js/reducers/sources.js

```diff
@@ -175,9 +177,58 @@ function getNewSelectedSource(state, id) : ?Source {
   return tabs.get(tabIndex + 1);
 }
 
-// We need this for tests, but I also think we should start bundling
-// in selectors into reducer modules, so we'll eventually start
-// exporting multiple things from here. Need to flesh out this idea.
-update.SourcesState = State;
+// Selectors
 
-module.exports = update;
+// Unfortunately, it's really hard to make these functions accept just
+// the state that we care about and still type if with Flow. The
+// problem is that we want to re-export all selectors from a single
+// module for the UI, and all of those selectors should take the
+// top-level app state, so we'd have to "wrap" them to automatically
+// pick off the piece of state we're interested in. It's impossible
+// (right now) to type those wrapped functions.
+type OuterState = { sources: Record<SourcesState> };
```

Comment:
Our coworker gozala (uses flow in browser.html) may have come up with a solution. But yeah, for now I gotta stop spending time on it..